packages/react-events/src/dom/Tap.js

```diff
@@ -0,0 +1,615 @@
+  if (onTapStart != null) {
+    const payload = {...state.gestureState, type};
+    dispatchDiscreteEvent(context, payload, onTapStart);
+  }
+}
+
+function dispatchChange(
+  context: ReactDOMResponderContext,
+  props: TapProps,
+  state: TapState,
+): void {
+  const onTapChange = props.onTapChange;
+  if (onTapChange != null) {
+    const payload = state.isActive;
+    dispatchDiscreteEvent(context, payload, onTapChange);
+  }
+}
+
+function dispatchUpdate(
+  context: ReactDOMResponderContext,
+  props: TapProps,
+  state: TapState,
+) {
+  const type = 'tap:update';
+  const onTapUpdate = props.onTapUpdate;
+  if (onTapUpdate != null) {
+    const payload = {...state.gestureState, type};
+    dispatchUserBlockingEvent(context, payload, onTapUpdate);
+  }
+}
+
+function dispatchEnd(
+  context: ReactDOMResponderContext,
+  props: TapProps,
+  state: TapState,
+): void {
+  const type = 'tap:end';
+  const onTapEnd = props.onTapEnd;
+  if (onTapEnd != null) {
+    const payload = {...state.gestureState, type};
+    dispatchDiscreteEvent(context, payload, onTapEnd);
+  }
+}
+
+function dispatchCancel(
+  context: ReactDOMResponderContext,
+  props: TapProps,
+  state: TapState,
+): void {
+  const type = 'tap:cancel';
+  const onTapCancel = props.onTapCancel;
+  if (onTapCancel != null) {
+    const payload = {...state.gestureState, type};
+    dispatchDiscreteEvent(context, payload, onTapCancel);
+  }
+}
+
+/**
+ * Responder implementation
+ */
+
+const responderImpl = {
+  targetEventTypes,
+  getInitialState(): TapState {
+    return createInitialState();
+  },
+  onMount(context: ReactDOMResponderContext, props: TapProps, state: TapState) {
+    // touchAction
+  },
+  onEvent(
+    event: ReactDOMResponderEvent,
+    context: ReactDOMResponderContext,
+    props: TapProps,
+    state: TapState,
+  ): void {
+    if (props.disabled) {
+      removeRootEventTypes(context, state);
+      if (state.isActive) {
+        dispatchCancel(context, props, state);
+        state.isActive = false;
+      }
+      return;
+    }
+
+    const nativeEvent: any = event.nativeEvent;
+    const eventTarget: Element = nativeEvent.target;
+    const eventType = event.type;
+
+    switch (eventType) {
+      // START
+      case 'pointerdown':
+      case 'mousedown':
+      case 'touchstart': {
+        if (!hasPointerEvents) {
+          if (eventType === 'mousedown' && state.ignoreEmulatedEvents) {
+            return;
+          }
+        }
+
+        if (!state.isActive && shouldActivate(event)) {
+          const eventPointerId = event.pointerId;
+          // Make mouse and touch pointers consistent.
+          if (eventPointerId != null) {
+            // Flow bug: https://github.com/facebook/flow/issues/8055
+            // $FlowExpectedError
+            eventTarget.releasePointerCapture(eventPointerId);
+          }
+
+          state.isActive = true;
+          state.buttons = nativeEvent.buttons;
+          state.pointerId = eventPointerId;
```

Comment:
Probably make a utility to populate this, as each of the 3 event types is different w.r.t. pointer ids